container/dangerzone.py

```diff
@@ -333,191 +269,99 @@ def pixels_to_pdf(self):
+        self.update_progress(f"Merging {num_pages} pages into a single PDF")
         args = ["pdfunite"]
         for page in range(1, num_pages + 1):
             args.append(f"/tmp/page-{page}.pdf")
         args.append(f"/tmp/safe-output.pdf")
-        try:
-            p = subprocess.run(
-                args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=60
-            )
-        except subprocess.TimeoutExpired:
-            self.output(
-                True,
-                "Error merging pages into a single PDF, pdfunite timed out after 60 seconds",
-                percentage,
-            )
-            return 1
-        if p.returncode != 0:
-            self.output(
-                True,
-                "Merging pages into a single PDF failed",
-                percentage,
-            )
-            return 1
+        run_command(
+            args,
+            error_message="Merging pages into a single PDF failed",
+            timeout_message=f"Error merging pages into a single PDF, pdfunite timed out after {DEFAULT_TIMEOUT} seconds",
+        )
 
-        percentage += 2
+        self.percentage += 2
 
         # Compress
-        self.output(
-            False,
-            f"Compressing PDF",
-            percentage,
-        )
+        self.update_progress(f"Compressing PDF")
         compress_timeout = num_pages * 3
-        try:
-            p = subprocess.run(
-                ["ps2pdf", "/tmp/safe-output.pdf", "/tmp/safe-output-compressed.pdf"],
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
-                timeout=compress_timeout,
-            )
-        except subprocess.TimeoutExpired:
-            self.output(
-                True,
-                f"Error compressing PDF, ps2pdf timed out after {compress_timeout} seconds",
-                percentage,
-            )
-            return 1
-        if p.returncode != 0:
-            self.output(
-                True,
-                f"Compressing PDF failed",
-                percentage,
-            )
-            return 1
+        run_command(
+            ["ps2pdf", "/tmp/safe-output.pdf", "/tmp/safe-output-compressed.pdf"],
+            timeout_message=f"Error compressing PDF, ps2pdf timed out after {compress_timeout} seconds",
+            error_message="Compressing PDF failed",
+            timeout=compress_timeout,
+        )
 
-        percentage = 100.0
-        self.output(False, "Safe PDF created", percentage)
+        self.percentage = 100.0
+        self.update_progress("Safe PDF created")
 
         # Move converted files into /safezone
         shutil.move("/tmp/safe-output.pdf", "/safezone")
         shutil.move("/tmp/safe-output-compressed.pdf", "/safezone")
 
-        return 0
-
-    def output(self, error, text, percentage):
-        print(json.dumps({"error": error, "text": text, "percentage": int(percentage)}))
+    def update_progress(self, text, *, error: bool = False):
```

Comment:
The unlabelled `, True` and `, False` values in the function calls were just a bit unclear, so this gives it a label when used, or omits it by default.

Reply:
I see your point. Basically, It's like a linting thing that forces one to use `error` as keyword argument instead of a positional one, thus preventing more confusing code like `update_progress("encountered a problem", True)`.

Sounds good.